### PR TITLE
fix(DB/Fishing): Implemented day/night fishing events.

### LINKED ISSUE #7984: Various Fish Not Following Day/Night Catching Restrictions
### What client do you play on?

enUS

### Faction

- [X] Alliance
- [X] Horde

### Content Phase:

- [ ] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [X] 40-49
- [ ] 50-59

### Current Behaviour
Original report: https://github.com/chromiecraft/chromiecraft/issues/1777

[Raw Nightfin Snapper](https://wowgaming.altervista.org/aowow/?item=13759) and [Raw Sunscale Salmon](https://wowgaming.altervista.org/aowow/?item=13760) are completely ignoring the server clock and just dropping whenever you fish in level 40-49 zones.

Spend 2 hours fishing (around 1 am to 3 am server time), and I left with twice Sunscale Salmons than Nightfin Snapper, where I was supposed to get very very few or none at all.

Tester: confirmed that the Nightfins can be caught during the daylight phase, and the Sunscales during the night phase:

Server time is 3:01 AM = full night - Sunscale caught:
WoWScrnShot_091921_155351

Server time is 3:54 PM = full day - Nightfin caught:
WoWScrnShot_091921_155447

These fish are both in RLTs 11008 and 11010, but have identical drop rates. If there's a specific RLT or FLT for daylight or nighttime fishing, I haven't found it, or any mechanism for swapping between them based on server time.

### Expected Blizzlike Behaviour

[This Wowhead comment has the best description of how it's supposed to be](https://www.wowhead.com/item=13759/raw-nightfin-snapper#comments:id=63623)
> from WoWwiki
> http://www.wowwiki.com/Fishing
>
> There are 4 separate intervals of fishing times. These intervals are 6am-12pm, 12pm-6pm, 6pm-12am, and 12am-6am. During this time the drop rates of certain fish varies, for example, Sunscale Salmon and Nightfin Snapper. According to experimentation, the cycle appears to be the following:
> 
> * 6am-12pm - Nightfin and Sunscale
> * 12pm-6pm - Sunscale only
> * 6pm-12am - Nightfin and Sunscale
> * 12am-6am - Nightfin only
> 
> Nightfin Snapper has the highest drop rate from 12am-6am, but is never found between 12pm-6pm. On the other hand, Sunscale Salmon is never found between 12am-6am, but drops quite well from 12pm-6pm.

### Source

https://www.wowhead.com/item=13759/raw-nightfin-snapper#comments
https://wowwiki-archive.fandom.com/wiki/Raw_Nightfin_Snapper?oldid=2376277
https://www.wowhead.com/item=13760/raw-sunscale-salmon#comments
https://wowwiki-archive.fandom.com/wiki/Raw_Sunscale_Salmon?oldid=2040656
https://wowwiki-archive.fandom.com/wiki/Nocturnal

### Steps to reproduce the problem

Manually set the server time to a time in one of the phases, i.e. midnight to 6AM = full night, or midday to 6PM = full day. Then fish in a 40-ish zone (I went to Camp Mojache, but lots of other zones will work) and fish. Then look for the out-of-phase fish, which shouldn't take too long as their drop rate is 17-18%.

### Extra Notes

_No response_

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/7e2e6f8ee85f82f57a3c6e0f6f645b4c42292e7a

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)

### Customizations

None

### Server

ChromieCraft

### PR DESCRIPTION
**Pull request by UltraNix**
Fixes #7984

## Issues Addressed:

- Closes #7984
- Closes https://github.com/chromiecraft/chromiecraft/issues/1777

## Tests Performed:

- Not tested

## How to Test the Changes:

More info [here](#7984)

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Asthalor** [comment]
Tested:

On a 12:03 AM server time: 

image

image

Nightfin should not be available i think.

**Comment by UltraNix** [comment]
Have you tried on 01.05 PM? Maybe it's because of timezones.

**Comment by Nyeriah** [comment]
@azerothcore/testers could someone check this? Thanks!

**Comment by acidmanifesto** [comment]
This works as intended and is aligned with TC as well looking at their db as well.

File: data/sql/updates/pending_db_world/rev_1633020417816915100.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1633020417816915100');
+
+DELETE FROM `game_event` WHERE `eventEntry` IN (79,80);
+INSERT INTO `game_event` (`eventEntry`,`start_time`,`end_time`,`occurence`,`length`,`holiday`,`description`,`world_event`, announce) VALUES
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1633020417816915100.sql:4]:
```suggestion
INSERT INTO `game_event` (`eventEntry`,`start_time`,`end_time`,`occurence`,`length`,`holiday`,`description`,`world_event`, `announce`) VALUES
```

**Review by Nefertumm — APPROVED** [review]

**Pull request merged**